float64 x
float64 y
float64 distance
float64 heading
#float64 correctedX
#float64 correctedY
int32 id